From fabcc09cd4cd3b6417e0d56888b386f95b485b24 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 19 Oct 2023 10:51:55 +0800
Subject: [PATCH 055/206] mtk: wpa_supplicant: force MLD STA to use SAE H2E
 during authentication

Otherwise the MLD STA setup will fail with hostapd MLD AP.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 wpa_supplicant/sme.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/sme.c b/wpa_supplicant/sme.c
index d4e9cebbc..b2ad6e76b 100644
--- a/wpa_supplicant/sme.c
+++ b/wpa_supplicant/sme.c
@@ -207,7 +207,8 @@ static struct wpabuf * sme_auth_build_sae_commit(struct wpa_supplicant *wpa_s,
 	if (wpa_key_mgmt_sae_ext_key(key_mgmt) &&
 	    sae_pwe != SAE_PWE_FORCE_HUNT_AND_PECK)
 		use_pt = 1;
-	if (bss && is_6ghz_freq(bss->freq) &&
+
+	if (bss && (is_6ghz_freq(bss->freq) || !is_zero_ether_addr(bss->mld_addr)) &&
 	    sae_pwe != SAE_PWE_FORCE_HUNT_AND_PECK)
 		use_pt = 1;
 #ifdef CONFIG_SAE_PK
-- 
2.45.2

